### PR TITLE
Odometry fixes and updates

### PR DESCRIPTION
**Pull request by PandaNinja10**
Yaw seting fix and Pathplanner cordinate converssion for shuffleboard display.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 1]
```diff
@@ -356,13 +360,10 @@ public void setChassisSpeeds(ChassisSpeeds chassisSpeeds, boolean isOpenLoop) {
...
-   * @param degrees the new yaw angle, in degrees.
+   * @param rotation the new yaw angle as Rotation2d
    */
-  public void setYaw(double degrees) {
-    // the odometry stores an offset from the current pigeon angle
-    // changing the angle makes that offset inaccurate, so must reset the pose as well.
-    // keep the same translation, but set the odometry angle to what we want the angle to be.
-    resetOdometry(new Pose2d(getPose().getTranslation(), Rotation2d.fromDegrees(degrees)));
+  public void setYaw(Rotation2d rotation) {
+    m_pigeon.setYaw(rotation.getDegrees());
```
Comment (vfurman-gh) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:366]: It is very confusing to have public methods setYaw(), setPigeonYaw(), getYaw() and getRawPigeonYaw() methods with slightly different semantics and no clear explanation of what they do. Can we reduce it to just a pair of public setXX/getXX? E.g., the setPigeonYaw could be called setYawForAlllance and moved to the caller or utility class.

File: src/main/java/frc/robot/util/Conversions.java [hunk 2]
```diff
@@ -113,4 +117,20 @@ public static double ElevatorHeightToExtension(double height) {
...
+     * 
+     * @param pose pose to convert
+     * @param alliance alliance PathPlanner is using for their origin
+     * @return converted pose
+     */
+    public static Pose2d absolutePoseToPathPlannerPose(Pose2d pose, Alliance alliance){
+        if (alliance == Alliance.Red){
+            return pose.relativeTo(new Pose2d(FieldConstants.kFieldLength, FieldConstants.kFieldWidth, new Rotation2d(Math.PI)));
+        }
+        return new Pose2d(pose.getX(), pose.getY(), new Rotation2d(pose.getRotation().getRadians()));
```
Comment (vfurman-gh) [inline src/main/java/frc/robot/util/Conversions.java:134]: why not` new Pose2d(..., ..., pose.getRotation())`?

**Comment by vfurman-gh** [comment]
I recommend to capture in a document all the transformations we do to coordinate origin, driver controls,  PathPlanner trajectories and whatever other objects when we go from Blue to Red. This will make life of code readers and future code maintainers much easier.

File: src/main/java/frc/robot/util/Conversions.java [hunk 3]
```diff
@@ -113,4 +117,20 @@ public static double ElevatorHeightToExtension(double height) {
     public static double ElevatorExtensionToHeight(double extension) {
       return (extension * Math.sin(ElevatorConstants.kElevatorAngle)) + ElevatorConstants.kElevatorBaseHeight;
     }
+
+    /**
+     * Converts between Absolute coordinate system pose of origin right of blue driver station
+     * to coordinate system where origin the current alliance driver station
+     * The transformation is reversible so there is no second function to convert back
```
Comment (vfurman-gh) [inline src/main/java/frc/robot/util/Conversions.java:124]: I think the proper name for this is "self-inverse" Reversible means that there exists a reverse transformation, here you mean that reverse transformation is identical to the original one.

File: src/main/java/frc/robot/commands/DefaultDriveCommand.java [hunk 4]
```diff
@@ -29,12 +29,12 @@ public void execute() {
...
     double slowRotFactor = m_driver.getIsSlowMode() ? DriveConstants.kSlowRotFactor : 1;
 
-    int reversedForRed = DriverStation.getAlliance() == Alliance.Blue ? -1 : 1;
+    int reversedForRed = DriverStation.getAlliance() == Alliance.Blue ? 1 : -1;
 
     /* Drive */
     m_swerve.drive(
       m_driver.getForwardTranslation() * slowFactor * reversedForRed,
-      m_driver.getSideTranslation() * slowFactor * reversedForRed,
+      m_driver.getSideTranslation() * slowFactor * -reversedForRed,
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/commands/DefaultDriveCommand.java:37]: this doesn't change anything. If you want the variable name to be accurate just change it to alianceReversal

File: src/main/java/frc/robot/commands/auto/PathPlannerCommand.java [hunk 5]
```diff
@@ -8,60 +8,69 @@
...
-            )
-        );
-    }
+    addCommands(new InstantCommand( () -> {
+      PathPlannerTrajectory path = PathPlannerTrajectory.transformTrajectoryForAlliance(
+        pathGroup.get(pathIndex),DriverStation.getAlliance());
+      if (resetPose) {
+        drive.setPigeonYaw(path); 
+        drive.resetOdometry(Conversions.absolutePoseToPathPlannerPose(
+        path.getInitialHolonomicPose(), DriverStation.getAlliance()));
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/commands/auto/PathPlannerCommand.java:61]: Isn't odometry using an absolute pose? And the input parameter into the conversion is a pathplanner pose? So it should be Conversions.pathPlannerPoseToAbsolutePose(), right?

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 6]
```diff
@@ -237,6 +236,10 @@ public Rotation2d getRoll() {
   * @return the pigeon's heading in a Rotation2d
   */
   public Rotation2d getYaw() {
+    return m_poseEstimator.getEstimatedPosition().getRotation();
+  }
+
+  public Rotation2d getRawPigeonYaw(){
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:242]: remember javadoc comments, but im not sure you need two functions for yaw

**Review by AnthonyFurman — APPROVED** [review]

**Pull request merged**